Mathlib/CategoryTheory/Limits/Constructions/Over/Products.lean

```diff
@@ -20,11 +23,97 @@ open CategoryTheory CategoryTheory.Limits
 
 variable {J : Type w}
 variable {C : Type u} [Category.{v} C]
-variable {X : C}
+variable {X Y Z : C}
+
+/-!
+## Binary products
+
+One could have used the following but it gives worse defeqs.
+`(Cones.postcomposeEquivalence (diagramIsoCospan _).symm).trans (conesEquiv _ (pair Y Z))`
```

Comment:
This should be `(Cones.postcomposeEquivalence (diagramIsoCospan _).symm).trans
    (Over.ConstructProducts.conesEquiv _ (pair (Over.mk f) (Over.mk g)))` instead, and it only applies to `pullbackConeEquivBinaryFan`.

For coproducts we unfortunately do not have the corresponding general results.